feat/About+gallery+companies sections

OmarQussas opened the pull request:
Develop About Us section with brand text and imagery, you need to make sure implement the same design as Figma

Build a Gallery section to showcase gifts with the same layout in Figma.

Add logos for trusted partner companies

OmarQussas committed:
about section

OmarQussas committed:
updated gallery section

- added gallery images
- created small section heading component to make it reusable
- updated tailwind config with spacings headings and paragraph to improve maintainability

OmarQussas committed:
companies section

- created gallery
- applied darkmode styles across the previous components

OmarQussas committed:
Merge branch 'dev' into About+Gallery+Companies-Sections

YoussefAbdelrazk reviewed (commented):
this code is good i see it but 
1- can move about ,companies and gallery in features folder 
2- there was error in run project can you check it again 
3- name branch will be feat/............

OmarQussas commented:
> this code is good i see it but 1- can move about ,companies and gallery in features folder 2- there was error in run project can you check it again 3- name branch will be feat/............

done ✅

OmarQussas committed:
updated locations

created the features folder to follow the folder structure
renamed the branch
fixed the error

OmarQussas committed:
Merge branch 'About+Gallery+Companies-Sections' of https://github.com/YoussefAbdelrazk/ROSE-APP into About+Gallery+Companies-Sections

MoaazHail commented on src/components/features/About/About.tsx:
```diff
@@ -0,0 +1,118 @@
+
+export default function About() {
+  return (
+    <main>
```
you should have only one main tag in html file and we have already one main tag in layout.

MoaazHail commented on src/components/features/Gallery/Gallery.tsx:
```diff
@@ -0,0 +1,78 @@
+
+export default function Gallery() {
+  return (
+    <Section>
```
In my opinion you must delete one of these two section tags to performance

MoaazHail reviewed (commented):
Actually, this code is very nice and clean ✨
Good job bro 💪

MoaazHail commented:
Everything looks good overall! 💪
Just a few small process-related notes (not about your code):
1- Make sure to add label in your PR.
2- Make sure to assign yourself to the PR.
3- Please add your issue next time.

All of these are great ✅

Shamsmedhat commented on public/assets/images/about/gift-img.svg:
Don’t upload pre-styled or cropped images.
Upload the full original image and use CSS/Tailwind to match the Figma layout this ensures flexibility and scalability.

Shamsmedhat commented on src/components/features/About/About.tsx:
```diff
@@ -0,0 +1,118 @@
+            </h2>
+
+            <p className='text-p-2 py-element-xs text-zinc-500 leading-none '>
+              Make every moment memorable with our premium gift boxes. Carefully
```
It’s better to use CSS (Tailwind classes like uppercase or capitalize) to control text casing instead of hardcoding the content in uppercase/lowercase. This way, any future adjustments become much easier and cleaner without needing to edit the actual content.

Shamsmedhat commented on src/components/features/About/About.tsx:
Wrong naming, use about (lowercase) instead of About.

Shamsmedhat commented on src/components/features/About/About.tsx:
```diff
@@ -0,0 +1,118 @@
+            <p className='text-p-2 py-element-xs text-zinc-500 leading-none '>
+              Make every moment memorable with our premium gift boxes. Carefully
+              curated and beautifully packaged, each box is filled with
+              handpicked items designed to impress. Whether it's for a birthday,
```
it`&apos;`s not it's.

Shamsmedhat commented on src/components/features/About/About.tsx:
```diff
@@ -0,0 +1,118 @@
+                <Check
+                  className='text-maroon-700 dark:text-soft-pink-400'
+                  size={16}
+                />{' '}
```
No need for this spaces, `{' '}`

Shamsmedhat commented on src/components/features/Companies/Companies.tsx:
```diff
@@ -0,0 +1,47 @@
+import { cn } from '../../../lib/utils/utils';
+import Highlight from '../../shared/highlight';
+import { Section } from '../../layout';
+import companyImage1 from '../../../../public/assets/images/companies/1.svg';
```
Import images directly inside the component instead of importing them at the top of the file.

Shamsmedhat commented on src/components/features/Companies/Companies.tsx:
```diff
@@ -0,0 +1,47 @@
+  ];
+  return (
+    <Section
+      className={cn(
```
No need to use `cn` here if we're not using JS logic inside the `className`.

Shamsmedhat commented on src/components/features/Companies/Companies.tsx:
```diff
@@ -0,0 +1,47 @@
+    >
+      <div className='flex flex-col gap-section-xs bg-maroon-50 dark:bg-zinc-700 p-section-xs rounded-xl'>
+        <h1 className='text-h-3 md:text-h-3 font-poppins font-semibold  text-maroon-700 dark:text-soft-pink-200  text-center'>
+          Trusted by over <Highlight>4.5k+</Highlight> companies
```
This part needs to be updated after the translation `(next-intl)` is implemented.

OmarQussas replied:
sure

Shamsmedhat commented on src/components/features/Companies/Companies.tsx:
```diff
@@ -0,0 +1,47 @@
+        <div className='flex  justify-center gap-8 '>
+          {companyImages.map((src, i) => (
+            <Image
+              key={i}
```
React keys should be stable and unique identifiers. If the array changes order, gets filtered, or has items added/removed, index-based keys can cause rendering bugs and state issues.

**Better practices is** Instead of relying on indexes, create unique, stable identifiers for each element.

Shamsmedhat reviewed (changes requested):
Good job 👏, your UI is very close and looks great. You’re almost done, just a few enhancements are needed.

Branch naming is not correct.
Follow this rules.

![3da10e87-e2cf-4ed3-9fb7-c7f593f7fabf](https://github.com/user-attachments/assets/94088008-3253-4e80-8657-cd41ab071a2d)


Your UI Vs Figma.
<img width="992" height="538" alt="image" src="https://github.com/user-attachments/assets/4528c984-d449-4f5f-9ce4-b8e246f9b241" />

OmarQussas committed:
fix(ui/ux updates)

1 - used full images and cropped them by tailwind/CSS for flexibility
2- refactored the naming to lowercase (About --> about )
3- fixed the about - main image design to match with figma
4- removed the useless 'cn' utility and added an id other than the index for the map

Shamsmedhat reviewed (approved):
Good job 👏
